Tools/Taxonomy/OwaspMobileTop10/OwaspMobileTop10TaxonomyGenerator.cs

```diff
@@ -0,0 +1,133 @@
+
+                Run run = this.ConvertToSarif(results, version);
+
+                SarifLog log = new SarifLog
+                {
+                    Runs = new Run[] { run }
+                };
+
+                File.WriteAllText(targetFilePath, JsonConvert.SerializeObject(log, Formatting.Indented));
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+                return false;
+            }
+
+            return true;
+        }
+
+        private List<OwaspMobileTop10Record> ReadFromMd(string folderPath)
+        {
+            List<OwaspMobileTop10Record> records = new List<OwaspMobileTop10Record>();
+            DirectoryInfo d = new DirectoryInfo(folderPath);
+            FileInfo[] files = d.GetFiles("m*-*.md");
+            files = files.OrderBy(f => int.Parse(Regex.Match(f.Name, @"\d+").Value)).ToArray();
+
+            foreach (FileInfo file in files)
+            {
+                List<string> allHeaders = new List<string>();
+                string text = File.ReadAllText(file.FullName);
+                MarkdownDocument doc = Markdig.Markdown.Parse(text);
+                var headerBlocks = doc.Where(d => d.GetType() == typeof(HeadingBlock)).ToList();
+                string titleText = headerBlocks.Select(h => ((LeafBlock)h).Inline.FirstChild.ToString().Trim()).First();
+                Match match = Regex.Match(titleText, "title: \"(?<id>.*): (?<name>.*)\"");
+                var headerTexts = headerBlocks.Select(h => ((LeafBlock)h).Inline.FirstChild.ToString().Trim()).Skip(1).ToList();
+                allHeaders.AddRange(headerTexts);
+                records.Add(new OwaspMobileTop10Record()
+                {
+                    Id = match.Groups["id"].Value,
+                    Name = match.Groups["name"].Value,
+                    ShortDescription = string.Join("\r\n", allHeaders)
+                });
+            }
+
+            return records;
+        }
+
+        private Run ConvertToSarif(List<OwaspMobileTop10Record> records, string version)
+        {
+            IList<ToolComponent> taxonomies = new List<ToolComponent>();
+            ToolComponent toolComponent = new ToolComponent
+            {
+                Version = version,
+                Name = version switch
+                {
+                    "2014" => Constants.OwaspMobileTop102014.Name,
+                    "2016" => Constants.OwaspMobileTop102016.Name,
+                    _ => throw new NotImplementedException()
+                },
+                Guid = version switch
+                {
+                    "2014" => Constants.OwaspMobileTop102014.Guid,
+                    "2016" => Constants.OwaspMobileTop102016.Guid,
+                    _ => throw new NotImplementedException()
+                },
+                InformationUri = new Uri(version switch
+                {
+                    "2014" => Constants.OwaspMobileTop102014.Location,
```

Comment:
>Constants.OwaspMobileTop102014.Location [](http://example.com/codeflow?start=30&length=39)

normally, this is a hard-coded location

Reply:
pls, check the other taxonomies

Reply:
yes I can see in Yong's PR this Location is the location of .sarif file not the location of the source.
Fixed.